[2025-08-08 Fiddler](https://thefiddler.substack.com/p/can-you-canoodle-at-the-coldplay)
====================

Fiddler
-------
The chance of getting caught at a single concert is 0.5%, so the chance of not getting
caught in $n$ concerts is $0.995^n$.  So if $0.995^n = 0.5$, $n = \log 0.5/\log 0.995$,

In [1]:
log(0.5)/log(0.995)

138.282572986061

So we can go to 138 concerts with less than even chances of getting caught.

Extra credit
------------

The probability distribution of individuals as a function of the fraction is
$p_I(f) = 1$ when $0 \le f \le 1$.

The probability that an individual with a fraction $x$ is in a couple with
fraction of at least $c$ is $1-c/x$ when $x > c$ and $0$ when $x \le c$.  (More
generally, $H(x-c)\int_{c/x}^1 dy\,p_I(y)$.)

Integrating over $x$, the probability that a couple's fraction is at least $c$
is $\int_c^1 dx\,(1-c/x) = 1 - c + c\log c$.  (In general,
$\int_0^1 p_I(x)H(x-c)\int_{c/x}^1 dy\,p_I(y)$.)

The probability distribution of couples is proportional to the derivative of the
probability that a couple's fraction is less than $c$, $d/dc(c-c\log c) = -\log c$.

The probability distribution of couples canoodling at any instant in time is
proportional to $-c\log c$, which has a maximum at $-\log C - 1 = 0$, or $C = 1/e$.

So, at any instant in time, there should be more couples on average with fraction
$1/e$ than any other fraction.

Simulations
-----------
Simulations show that the largest number of couples have a fraction somewhere around 0.35-0.37,
which agrees with $1/e \approx 0.3679$.

    $ go run 20250808.go
    0.000000: 4035
    0.010000: 10230
    0.020000: 14620
    0.030000: 18831
    0.040000: 22306
    0.050000: 25438
    0.060000: 28177
    0.070000: 30913
    0.080000: 33379
    0.090000: 35657
    0.100000: 37613
    0.110000: 39586
    0.120000: 41413
    0.130000: 43173
    0.140000: 44486
    0.150000: 46255
    0.160000: 47121
    0.170000: 48764
    0.180000: 49809
    0.190000: 51005
    0.200000: 52182
    0.210000: 52667
    0.220000: 53471
    0.230000: 54768
    0.240000: 55040
    0.250000: 55835
    0.260000: 56340
    0.270000: 56308
    0.280000: 57371
    0.290000: 57515
    0.300000: 57884
    0.310000: 58191
    0.320000: 58240
    0.330000: 58736
    0.340000: 58702
    0.350000: 58799
    0.360000: 58493
    0.370000: 58774
    0.380000: 58506
    0.390000: 58376
    0.400000: 58059
    0.410000: 58274
    0.420000: 58052
    0.430000: 57921
    0.440000: 57840
    0.450000: 57461
    0.460000: 57013
    0.470000: 56575
    0.480000: 55665
    0.490000: 55882
    0.500000: 55538
    0.510000: 55026
    0.520000: 53821
    0.530000: 53723
    0.540000: 52695
    0.550000: 52169
    0.560000: 51523
    0.570000: 51133
    0.580000: 50260
    0.590000: 49224
    0.600000: 48902
    0.610000: 48021
    0.620000: 47206
    0.630000: 46252
    0.640000: 45868
    0.650000: 44518
    0.660000: 43142
    0.670000: 42368
    0.680000: 41695
    0.690000: 40259
    0.700000: 39437
    0.710000: 38386
    0.720000: 37169
    0.730000: 36627
    0.740000: 35142
    0.750000: 33754
    0.760000: 32861
    0.770000: 31404
    0.780000: 30394
    0.790000: 29235
    0.800000: 28112
    0.810000: 26808
    0.820000: 25882
    0.830000: 24101
    0.840000: 22815
    0.850000: 21383
    0.860000: 19781
    0.870000: 18688
    0.880000: 17353
    0.890000: 15932
    0.900000: 14360
    0.910000: 13108
    0.920000: 11718
    0.930000: 9899
    0.940000: 8527
    0.950000: 7084
    0.960000: 5437
    0.970000: 4001
    0.980000: 2271
    0.990000: 834


Making the $\lfloor$rounds$\rceil$
----------------------------------
For 2,

    AB
    BA

For 3, it's not possible, since each ordered pair either does not appear or appears twice.

    ABC   ABC
    BCA   CAB
    CAB   BCA

For 4, 3 different ones can be made.

    ABCD    ABCD    ABCD
    BDAC    CADB    DCBA
    CADB    DCBA    BDAC
    DCBA    BDAC    CADB